### PR TITLE
break recursion when calling GetIAMCredentials

### PR DESCRIPTION
**Pull request by rallister**
### Relevant Issue (if applicable)
https://github.com/s3fs-fuse/s3fs-fuse/issues/1196

### Details
Added a flag to not insert authHeaders when calling GetIAMCredentials, because they are not needed in this call but also to break recursion which occurs in issue 1196
CheckIAMCredentialUpdate() -> 
   GetIAMCredentials() -> 
       RequestPerform() ->
          insertAuthHeaders() ----> back to CheckIAMCredentialUpdate()

You might come up with a better way but this quickly got me out of a bind.

### CODE REVIEW

**Comment by ggtakec** [comment]
@rallister Thanks for your great help.
I think that this PR code can fix the problem (loop).
I'd like to make minor changes to the code, so I'll comment and check it out. (Trivial description change)

File: src/curl.cpp [hunk 1]
```diff
@@ -2384,7 +2384,9 @@ int S3fsCurl::RequestPerform()
   long responseCode;
   int result        = S3FSCURL_PERFORM_RESULT_NOTSET;
 
-  insertAuthHeaders();
+  if(!dontAddAuthHeaders)
+     insertAuthHeaders();
+
```
Comment (ggtakec) [inline src/curl.cpp]: Please enclose the condition sentence in curly braces.

**Comment by ggtakec** [comment]
@rallister Maybe you might fail with MacOS test on Travis.
Now I merged into master to remove the error.
If you would get an error, please build again.

**Comment by rallister** [comment]
commits simply refuse to squash.

**Comment by rallister** [comment]
@ggtakec ok it's done.

**Comment by ggtakec** [comment]
@rallister Thanks and I merged this PR.

**Pull request merged**